tests/unit/components/test_braveadblock.py

```diff
@@ -95,47 +95,34 @@ def run_function_on_dataset(given_function):
     contains tuples of (url, source_url, type) in each line. We give these
     to values to the given function, row by row.
     """
-
-    def dataset_type_to_enum(type_int: int) -> ResourceType:
-        """Translate the dataset's encoding of a resource type to Qutebrowser's."""
-        if type_int == 0:
-            return ResourceType.unknown
-        elif type_int == 1:
-            return ResourceType.image
-        elif type_int == 2:
-            return ResourceType.stylesheet
-        elif type_int == 3:
-            return ResourceType.media
-        elif type_int == 4:
-            return ResourceType.script
-        elif type_int == 5:
-            return ResourceType.font_resource
-        elif type_int == 6:
-            return ResourceType.xhr
-        else:
-            assert type_int == 7
-            return ResourceType.sub_frame
+    dataset_type_to_enum = {
+        0: ResourceType.unknown,
+        1: ResourceType.image,
+        2: ResourceType.stylesheet,
+        3: ResourceType.media,
+        4: ResourceType.script,
+        5: ResourceType.font_resource,
+        6: ResourceType.xhr,
+        7: ResourceType.sub_frame,
+    }
```

Comment:
One thing I don't quite get yet: Where are those integers coming from? I can see them in `tests/end2end/data/adblock_dataset.tsv.gz`, but where's that coming from? Is it from some other project?

Perhaps it'd be good to move those files into `tests/end2end/data/adblock/` and add a `README` with the respective attribution (and probably also licenses)?

Reply:
I took the data from https://github.com/brave/adblock-rust/blob/master/data/ublock-matches.tsv and ran it through some preprocessing scripts I wrote. One of them converted request type strings to ints:
```
def type_to_int(type_str: str) -> Optional[int]:
    if type_str == "other":
        return 0
    elif type_str == "image":
        return 1
    elif type_str == "stylesheet":
        return 2
    elif type_str == "media":
        return 3
    elif type_str == "script":
        return 4
    elif type_str == "font":
        return 5
    elif type_str == "xmlhttprequest":
        return 6
    elif type_str == "sub_frame":
        return 7
```
At first we didn't compress the file with GZip, so using integers was meant to save space.

Reply:
Ah! I think it'd be good to undo that change then, if you can - that makes the tests a bit easier to understand and keeps the file closer to the original.

It's always great if testcases from upstream projects can be reused, but I think what I said above definitely applies then - this should go into a folder with a README, the [appropriate LICENSE file](https://github.com/brave/adblock-rust/blob/master/LICENSE) and ideally your preprocessing scripts included as well. Those don't need to be cleaned up much, they'd mainly serve as a starting point if someone wants to e.g. update that file in the future.